praktikum/poglavlja/12_dinamicka_alokacija/primeri/realloc.c

```diff
@@ -0,0 +1,70 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+int main()
+{
+    int *a, n;
+
+    do
+    {
+        printf("Unesite duzinu niza: ");
+        scanf("%d", &n);
+
+    } while(n <= 0);
+
+    a = (int *) malloc(n * sizeof(int));
+
+    if (a == NULL)
+    {
+        printf("Memorija se ne moze zauzeti.\n");
+        exit(EXIT_FAILURE);
+    }
+
+    for(int i = 0; i < n; i++)
+    {
+        printf("a[%d] = ", i);
+        scanf("%d", &a[i]);
+    }
+
+    int m;
+
+    do
+    {
+        printf("Unesite novu duzinu niza: ");
+        scanf("%d", &m);
+
+    } while(m <= 0);
+
+    // Pokušaj da se poveća veličina niza koristeći realloc
+    void *tmp = (int *) realloc(a, m * sizeof(int));
```

Comment:
Zašto ovde tip ``void *``, a ne ``int *``? Onda je kasnije kastovanje nepotrebno.

Reply:
popravljeno